presto-main/src/main/java/com/facebook/presto/operator/ArrayRowUnnester.java

```diff
@@ -0,0 +1,63 @@
+package com.facebook.presto.operator;
+
+import com.facebook.presto.spi.PageBuilder;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.block.ColumnarRow;
+import com.facebook.presto.spi.type.RowType;
+import com.facebook.presto.spi.type.Type;
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+
+public class ArrayRowUnnester
+    implements Unnester
+{
+    private final List<Type> fieldTypes;
+    private ColumnarRow block;
```

Comment:
What about just call it `columnarRow`?

Reply:
Sure, both makes sense. I don't really care.